apps/evm/lib/evm/message_call.ex

```diff
@@ -0,0 +1,129 @@
+defmodule EVM.MessageCall do
+  alias EVM.{ExecEnv, Memory, VM, Functions, Stack}
+  alias EVM.Interface.AccountInterface
+
+  @moduledoc """
+  Describes a message call function that used for all call opertations (call, delegatecall, callcode, staticcall).
+  """
+  defstruct [
+    :current_exec_env,
+    :current_machine_state,
+    :output_params,
+    # s
+    :sender,
+    # o
+    :originator,
+    # r
+    :recipient,
+    # c
+    :code_owner,
+    # p
+    :gas_price,
+    # v
+    :value,
+    # v with overline
+    :execution_value,
+    # d
+    :data,
+    # e
+    :stack_depth
+  ]
+
+  @doc """
+  Message call function. Described as Θ in the Eq.(98) of the Yellow Paper
+  """
+  def call(message_call) do
+    if enough_gas?(message_call) && valid_stack_depth?(message_call) do
+      execute(message_call)
+    else
+      failed_call(message_call)
+    end
+  end
+
+  defp enough_gas?(message_call) do
+    sender_balance =
+      AccountInterface.get_account_balance(
+        message_call.current_exec_env.account_interface,
+        message_call.sender
+      )
+
+    sender_balance >= message_call.execution_value
+  end
+
+  defp valid_stack_depth?(message_call) do
+    message_call.stack_depth < Functions.max_stack_depth()
+  end
+
+  defp execute(message_call) do
+    # first transitional state
+    message_call = transfer_gas_to_recipient(message_call)
+
+    message_call
+    |> prepare_call_execution_env()
+    |> execute_call(message_call)
+    |> update_state(message_call)
+  end
+
+  def transfer_gas_to_recipient(message_call) do
```

Comment:
The function is named `transfer_gas_to_recepient`, but as I can see you're transferring `wei` from the current sender to the "recipient"

Reply:
@vyorkin I guess you a mistake